x/execution/internal/keeper/keeper.go

```diff
@@ -99,31 +99,53 @@ func (k *Keeper) Create(ctx sdk.Context, msg types.MsgCreateExecution) (*executi
 		msg.Request.Inputs,
 		msg.Request.Tags,
 		msg.Request.ExecutorHash,
+		msg.Request.Confs,
 	)
 	exec.BlockHeight = ctx.BlockHeight()
 
 	store := ctx.KVStore(k.storeKey)
-	if store.Has(exec.Hash) {
-		return nil, fmt.Errorf("execution %q already exists", exec.Hash)
-	}
-	if err := exec.Execute(); err != nil {
-		return nil, err
-	}
-	value, err := k.cdc.MarshalBinaryLengthPrefixed(exec)
-	if err != nil {
-		return nil, err
+	execExist := store.Has(exec.Hash)
+
+	if execExist {
+		if err := k.cdc.UnmarshalBinaryLengthPrefixed(store.Get(exec.Hash), &exec); err != nil {
+			return nil, err
+		}
+
+		if exec.Status != executionpb.Status_Voting {
+			return nil, fmt.Errorf("execution %q already exists", exec.Hash)
+		}
+	} else {
+		value, err := k.cdc.MarshalBinaryLengthPrefixed(exec)
+		if err != nil {
+			return nil, err
+		}
+		store.Set(exec.Hash, value)
+
+		execAddress := sdk.AccAddress(crypto.AddressHash(exec.Hash))
+		if err := k.bankKeeper.SendCoins(ctx, msg.Signer, execAddress, price); err != nil {
+			return nil, err
+		}
 	}
 
-	if !ctx.IsCheckTx() {
-		M.InProgress.Add(1)
+	var count int64
+	if exec.Confs > 0 && execExist {
```

Comment:
don't understand why we only add the signer if the execution already exists. What happens to the first signer who actually paid more than the other because he wrote more data. This one is not added to the list and will not be rewarded for leading this execution.